clients/gitlabrepo/tarball.go

```diff
@@ -114,40 +122,85 @@ func (handler *tarballHandler) setup() error {
 func (handler *tarballHandler) getTarball() error {
 	url := fmt.Sprintf("%s/api/v4/projects/%d/repository/archive.tar.gz?sha=%s",
 		handler.repourl.Host(), handler.repo.ID, handler.commitSHA)
+
+	// Create a temp file.  This automatically appends a random number to the name.
+	tempDir, err := os.MkdirTemp("", repoDir)
+	if err != nil {
+		return fmt.Errorf("os.MkdirTemp: %w", err)
+	}
+	repoFile, err := os.CreateTemp(tempDir, repoFilename)
+	if err != nil {
+		return fmt.Errorf("%w io.Copy: %v", errTarballNotFound, err)
+	}
+	defer repoFile.Close()
+	err = handler.apiFunction(url, tempDir, repoFile)
+	if err != nil {
+		return fmt.Errorf("gitlab.apiFunction: %w", err)
+	}
+	// Gitlab url for pulling combined ci
+	url = fmt.Sprintf("%s/api/v4/projects/%d/ci/lint",
+		handler.repourl.Host(), handler.repo.ID)
+	ciFile, err := os.CreateTemp(tempDir, "gitlabscorecard_lint*.json")
+	if err != nil {
+		return fmt.Errorf("os.CreateTemp: %w", err)
+	}
+	err = handler.apiFunction(url, tempDir, ciFile)
+	if err != nil {
+		return fmt.Errorf("gitlab.apiFunction: %w", err)
+	}
+	byteValue, err := os.ReadFile(ciFile.Name())
+	if err != nil {
+		return fmt.Errorf("os.ReadFile: %w", err)
+	}
+	var result gitLabLint
+	err = json.Unmarshal(byteValue, &result)
+	if err != nil {
+		return fmt.Errorf("json.Unmarshal: %w", err)
+	}
+
+	ciYaml, err := os.Create(tempDir + "/gitlabscorecard_flattened_ci.yaml")
```

Comment:
Where is the flattening happening? Does L147 give us a single file or multiple files?

Reply:
* [Pulling of the raw linted file](https://github.com/kgangerlm/scorecard-gitlab/blob/gitlab-packaging/clients/gitlabrepo/tarball.go#L141-L147)
* [Extracting the merged yaml (flattened file)](https://github.com/kgangerlm/scorecard-gitlab/blob/gitlab-packaging/clients/gitlabrepo/tarball.go#L151-L166) and saving to the file.

gitlab documentation: https://docs.gitlab.com/ee/api/lint.html

Reply:
Just noticed what you were asking... There is a json result that contains a few related CI/CD linting elements/attributes. The field that I care about is the `merged_yaml`. This is being saved off as a yaml file (hardcoded name).

Reply:
Gotcha, thanks!